cassandra/src/main/scala/geotrellis/spark/io/cassandra/CassandraLayerDeleter.scala

```diff
@@ -16,36 +16,52 @@
 
 package geotrellis.spark.io.cassandra
 
-import geotrellis.spark.LayerId
-import geotrellis.spark.io._
+import java.math.BigInteger
 
+import com.datastax.driver.core.{BoundStatement, PreparedStatement}
+import geotrellis.spark.{Boundable, LayerId}
+import geotrellis.spark.io._
 import com.typesafe.scalalogging.LazyLogging
 import com.datastax.driver.core.querybuilder.QueryBuilder
 import com.datastax.driver.core.querybuilder.QueryBuilder.{eq => eqs}
+import geotrellis.spark.io.avro.AvroRecordCodec
+import spire.math.Interval
+import spire.implicits._
+import spray.json.JsonFormat
 
 import scala.collection.JavaConverters._
+import scala.collection.immutable.VectorBuilder
+import scala.reflect.ClassTag
 
 class CassandraLayerDeleter(val attributeStore: AttributeStore, instance: CassandraInstance) extends LazyLogging with LayerDeleter[LayerId] {
 
-  def delete(id: LayerId): Unit = {
+  def delete[K: AvroRecordCodec: Boundable: JsonFormat: ClassTag](id: LayerId): Unit = {
     try {
       val header = attributeStore.readHeader[CassandraLayerHeader](id)
-      instance.withSessionDo { session =>
-        val squery = QueryBuilder.select("key")
-          .from(header.keyspace, header.tileTable).allowFiltering()
-          .where(eqs("name", id.name))
-          .and(eqs("zoom", id.zoom))
+      val keyIndex = attributeStore.readKeyIndex[K](id)
```

Comment:
Still not thrilled with having to go out and fetch the `KeyIndex` here, but I've yet to noodle through another way to fetch the range information I need in order to assign a consistent `zoomBin`.